wcc-contentful-app/lib/wcc/contentful/app/middleware/publish_at.rb

```diff
@@ -31,5 +42,82 @@ def before(time)
 
       Time.zone.now <= Time.zone.parse(time)
     end
+
+    if defined?(ActiveJob::Base)
+      def maybe_drop_job(entry)
+        publish_at = entry.dig('fields', 'publishAt', 'en-US')
+        unpublish_at = entry.dig('fields', 'unpublishAt', 'en-US')
+
+        drop_job_at(publish_at, entry) if publish_at.present? && before(publish_at)
+        drop_job_at(unpublish_at, entry) if unpublish_at.present? && before(unpublish_at)
+      end
+
+      def drop_job_at(timestamp, entry)
+        ts = Time.zone.parse(timestamp)
+        Job.set(wait_until: ts + 1.second).perform_later(entry, store)
+      end
+
+      class Job < ActiveJob::Base
+        include Wisper::Publisher
+
+        self.queue_adapter = :async
+
+        def self.cache
+          @cache ||= Rails.cache
+        end
+        attr_writer :cache
+
+        def initialize(*arguments)
+          super
+
+          subscribe(WCC::Contentful::Events.instance, with: :rebroadcast)
```

Comment:
this looks like a good candidate for a mixin so that it doesn't get forgotten somewhere

Reply:
I'm not 100% sure what you're envisioning, but here's my stab:

```ruby
module WCC::Contentful::Events::Publisher

  def initialize(*arguments)
    super

    subscribe(WCC::Contentful::Events.instance, with: :rebroadcast)
  end

  included do
    include Wisper::Publisher
  end
end
```

Then you just include it instead of `Wisper::Publisher` - thoughts?